cmd/recording.go

```diff
@@ -0,0 +1,126 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+  	"encoding/json"
+  	"net/http"
+	"os"
+
+  	"github.com/spf13/cobra"
+)
+
+var path string = "/_appmap/record"
+
+func init() {
+	var (
+		recordingCmd = &cobra.Command {
+			Use:   "appmap",
+      			Short: "Manage AppMap recordings",
+      			Run: func(cmd *cobra.Command, args []string) {
+				cmd.Usage()
+			},
+		}
+		// Starts a new AppMap recording
+		// Status 200: new recording session was started successfully
+		// Status 409: an exisiting recording session is already in progress
+		// Returns no body
+		recordingStartCmd = &cobra.Command {
+			Use:   "start [url]",
+			Short: "Start a new AppMap recording session",
+			Args:  cobra.ExactArgs(1),
+			Run: func(cmd *cobra.Command, args []string) {
+				url := args[0] + path
+                		response, err := http.Post(url,"",nil)
+                		if err != nil {
+					fail(err)
+				}
+				// best practice to defer close response.body
+				// what if no body exists?
+				defer response.Body.Close()
```

Comment:
Once you get here, `response` and `response.Body` shouldn't be nil, even if the HTTP response body was empty. So, this `defer` will be ok